plugins/inputs/modbus/modbus.go

```diff
@@ -662,6 +667,10 @@ func scaleFloat32(s float64, v float32) float32 {
 	return float32(float64(v) * s)
 }
 
+func scaleFloat64(s float64, v float64) float64 {
+	return float64(float64(v) * s)
```

Comment:
Everything's a float64 here, so just 
```suggestion
	return v * s
```